compat/mingw.c

```diff
@@ -1097,6 +1105,22 @@ int mingw_lstat(const char *file_name, struct stat *buf)
 		filetime_to_timespec(&(fdata.ftLastAccessTime), &(buf->st_atim));
 		filetime_to_timespec(&(fdata.ftLastWriteTime), &(buf->st_mtim));
 		filetime_to_timespec(&(fdata.ftCreationTime), &(buf->st_ctim));
+		if (!S_ISLNK(buf->st_mode))
+		{
+			HANDLE hf = CreateFileW(
```

Comment:
Hmm. I am a bit wary of this. The `CreateFileW()` call is _quite_ costly: it takes a _lot_ of time to execute (and `lstat()` is exercised by Git _a lot_, therefore that cost is quite noticeable on large code bases). That's why we decided to go with the much cheaper `GetFileAttributesExW()` call, even if it does not allow us to fill in all the `stat()` information. It is the reason why we do not have correct values for `nlink` and why we use the CreationTime instead of the ChangeTime, for example.

Reply:
> how to add files while keep the project compiles.

Add them to `compat/win32/`, then `git add` them, then mention them [in `config.mak.uname`](https://github.com/git-for-windows/git/blob/90ec7986f620ce96ea93bb367a50e22171b94f8d/config.mak.uname#L486-L491) and [in `CMakeLists.txt`](https://github.com/git-for-windows/git/blob/v2.41.0-rc2.windows.1/contrib/buildsystems/CMakeLists.txt#L299-L311).

> how to add a setting to conditionally enable this feature at runtime.

Simply follow the example of [the long paths support](https://github.com/git-for-windows/git/commit/e28db144e820c4599878def4175628d5e2de9726), in particular the config documentation and the `are_long_paths_enabled()` function.